src/jwk.rs

```diff
@@ -318,6 +319,67 @@ impl JWK {
         key.params = key.params.to_public();
         key
     }
+
+    pub fn thumbprint(&self) -> Result<String, Error> {
+        let json = match &self.params {
+            Params::RSA(rsa_params) => {
+                let n = rsa_params.modulus.as_ref().ok_or(Error::MissingModulus)?;
+                let e = rsa_params.exponent.as_ref().ok_or(Error::MissingExponent)?;
+                serde_json::json!({
+                    "e": String::from(e),
+                    "kty": "RSA",
+                    "n": String::from(n)
+                })
+            }
+            Params::OKP(okp_params) => {
+                serde_json::json!({
+                    "crv": okp_params.curve.clone(),
+                    "kty": "OKP",
+                    "x": String::from(okp_params.public_key.clone())
+                })
+            }
+            Params::EC(ec_params) => {
+                let curve = ec_params.curve.as_ref().ok_or(Error::MissingCurve)?;
+                let x = ec_params.x_coordinate.as_ref().ok_or(Error::MissingPoint)?;
+                let y = ec_params.y_coordinate.as_ref().ok_or(Error::MissingPoint)?;
+                serde_json::json!({
+                    "crv": curve.clone(),
+                    "kty": "EC",
+                    "x": String::from(x),
+                    "y": String::from(y)
+                })
+            }
+            Params::Symmetric(sym_params) => {
+                let k = sym_params
+                    .key_value
+                    .as_ref()
+                    .ok_or(Error::MissingKeyValue)?;
+                serde_json::json!({
+                    "k": String::from(k),
+                    "kty": "oct"
+                })
+            }
+        };
+        let json_string = serde_json::to_string(&json)?;
```

Comment:
`serde_json` has the `preserve_order` feature to preserve the order of the keys, JWK Thumbprint needs a lexicographical order and you're already sorting the keys in the creation so it should be good.

Reply:
Adding an explanatory note: 1ab94f1db652236ad36ba2678cc9c50f8a55c701

Reply:
Oh I meant that you need to enable the feature, but then that will impact everything so I don't know what to do. I imagine the cost of the indexed map isn't big for us.

Reply:
Updated in 6dc6f2d1f6088851f212599f350b5a0ec0068fa2 to format manually, so that the feature doesn't have to be enabled (although we might need it elsewhere anyway). Added tests for the remaining key types: d9ea8b3230a38bf82e1228c88846037b39ab3437